src/main/kotlin/com/hitc/hitweconomyplugin/main/utils/CreditsUtils.kt

```diff
@@ -0,0 +1,28 @@
+package com.hitc.hitweconomyplugin.main.utils
+
+import com.hitc.hitweconomyplugin.main.core.EPlayer
+import com.hitc.hitweconomyplugin.main.core.GameType
+import com.hitc.hitweconomyplugin.main.core.Score
+
+object CreditsUtils {
+
+    fun addCredits(gameType: GameType, score: Int, ePlayer: EPlayer) : Int {
+        val scores = GameFileUtils.loadScores(ePlayer)
+        val filteredScores = MutableList<Score?>(0) { null }
```

Comment:
Do
```kotlin
val filteredScores = scores.filter { it.gameType == gameType }
```

instead of
```kotlin
        val filteredScores = MutableList<Score?>(0) { null }
        for (currentScore in scores) {
            if (currentScore?.gameType == gameType) {
                filteredScores.add(currentScore)
            }
        }
```